client/rpc/trace/src/lib.rs

```diff
@@ -901,24 +901,32 @@ where
 		let mut traces: Vec<_> =
 			moonbeam_client_evm_tracing::formatters::TraceFilter::format(proxy)
 				.ok_or("Fail to format proxy")?;
+
+		let eth_transactions_by_index: BTreeMap<u32, H256> = eth_transactions
+			.iter()
+			.map(|t| (t.transaction_index, t.transaction_hash))
+			.collect();
+
 		// Fill missing data.
-		for trace in traces.iter_mut() {
+		for trace in traces
+			.iter_mut()
+			.filter(|t| eth_transactions_by_index.contains_key(&t.transaction_position))
```

Comment:
Please use `filter_map` instead to get directly the tx hash, and return Some((trace, tx_hash)) or None